x-pack/plugins/observability/public/utils/is_alert_details_enabled.ts

```diff
@@ -19,11 +26,21 @@ export const isAlertDetailsEnabledPerApp = (
 
   const ruleTypeId = alert.fields[ALERT_RULE_TYPE_ID];
 
+  // The feature flags for alertDetails are not specific enough so we need to check
+  // the specific rule types to see if they should be blocked or not.
+  if (ALLOWED_RULE_TYPES.includes(ruleTypeId)) {
+    return true;
+  }
+
+  // Since we are moving away from feature flags, this code will eventually be removed.
   const appNameFromAlertRuleType =
     ruleTypeId === 'xpack.uptime.alerts.monitorStatus' ||
     ruleTypeId === 'xpack.uptime.alerts.tlsCertificate'
       ? 'uptime'
-      : (ruleTypeId.split('.')[0] as keyof ConfigSchema['unsafe']['alertDetails']);
+      : ruleTypeId.split('.')[0];
+  if (isUnsafeAlertDetailsFlag(appNameFromAlertRuleType)) {
+    return config.unsafe?.alertDetails[appNameFromAlertRuleType]?.enabled;
```

Comment:
@maryam-saeidi, this code already does what you suggested in our meeting today. Aka, keep the other rule types under the feature flag. 

And I think what we can do, and this is what @simianhacker wants to do, is:
- We have a flag for the entire App (all rules for that app).
- Once a specific rule is done we move it to the `ALLOWED_RULE_TYPES `

Reply:
From what I see, this is not applicable to APM rule types other than `transaction_duration`, right?

So we cannot enable other APM rule types in the development mode behind the feature flag anymore. My idea was to keep the feature flag for the rest of the APM rule types and incrementally remove them when each rule type is ready.